tests/core/test_printf_thread.c

```diff
@@ -0,0 +1,45 @@
+#include <assert.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "../../system/lib/libc/musl/include/features.h"
+#include "../../system/lib/libc/musl/src/internal/stdio_impl.h"
+#include "../../system/lib/libc/musl/src/internal/libc.h"
```

Comment:
IIRC including these musl-internal headers from code other than musl-internal code is not supported an no longer works with the musl upgrade I'm looking at landing soon.

Any way we can avoid this?  I guess not..  I can look into a workaround when I next loop back around.

Reply:
It can be avoided by removing those `printf` statements at the beginning of `main()`, not sure if those are useful.

Another testcase that appears to do this is `tests/pthread/test_pthread_locale.c`:
https://github.com/emscripten-core/emscripten/blob/900511b7ee9327efc5929735f605bebed6377127/tests/pthread/test_pthread_locale.c#L13